desc/integrals/bounce_integral.py

```diff
@@ -1,26 +1,862 @@
+    # real fft over poloidal since usually m > n
+    return rfft2(f, axes=(-1, -2), norm="forward")
+
+
+def _transform_to_clebsch(grid, nodes, f, is_reshaped=False):
+    """Transform to Clebsch spectral domain.
+
+    Parameters
+    ----------
+    grid : Grid
+        Tensor-product grid in (θ, ζ) with uniformly spaced nodes in
+        (2π × 2π) poloidal and toroidal coordinates.
+        Preferably power of 2 for ``grid.num_theta`` and ``grid.num_zeta``.
+    nodes : jnp.ndarray
+        Shape (L, M, N, 2) or (M, N, 2).
+        DESC coordinates (θ, ζ) sourced from the Clebsch coordinates
+        ``FourierChebyshevSeries.nodes(M,N,domain=(0,2*jnp.pi))``.
+    f : jnp.ndarray
+        Function evaluated on ``grid``.
+
+    Returns
+    -------
+    a : FourierChebyshevSeries
+        Spectral coefficients of f(α, ζ).
+
+    """
+    assert nodes.shape[-1] == 2
+    if not is_reshaped:
+        f = grid.meshgrid_reshape(f, "rtz")
+
+    M, N = nodes.shape[-3], nodes.shape[-2]
+    return FourierChebyshevSeries(
+        f=interp_rfft2(
+            # Interpolate to nodes in Clebsch space,
+            # which is not a tensor product node set in DESC space.
+            xq=nodes.reshape(*nodes.shape[:-3], M * N, 2),
+            f=f[..., jnp.newaxis, :, :],
+            axes=(-1, -2),
+        ).reshape(*nodes.shape[:-3], M, N),
+        domain=(0, 2 * jnp.pi),
+    )
+
+
+def _transform_to_clebsch_1d(grid, alpha, theta, B, N_B, is_reshaped=False):
+    """Transform to single variable Clebsch spectral domain.
+
+    Notes
+    -----
+    The field line label α changes discontinuously, so the approximation
+    g defined with basis function in (α, ζ) coordinates to some continuous
+    function f does not guarantee continuity between cuts of the field line
+    until full convergence of g to f.
+
+    Note if g were defined with basis functions in straight field line
+    coordinates, then continuity between cuts of the field line, as
+    determined by the straight field line coordinates (ϑ, ζ), is
+    guaranteed even with incomplete convergence (because the
+    parameters (ϑ, ζ) change continuously along the field line).
+
+    Do not interpret this as superior function approximation.
+    Indeed, if g is defined with basis functions in (α, ζ) coordinates, then
+    g(α=α₀, ζ) will sample the approximation to f(α=α₀, ζ) for the full domain in ζ.
+    This holds even with incomplete convergence of g to f.
+    However, if g is defined with basis functions in (ϑ, ζ) coordinates, then
+    g(ϑ(α=α₀,ζ), ζ) will sample the approximation to f(α=α₀ ± ε, ζ) with ε → 0 as
+    g converges to f.
+
+    (Visually, the small discontinuity apparent in g(α, ζ) at cuts of the field
+    line will not be visible in g(ϑ, ζ) because when moving along the field line
+    with g(ϑ, ζ) one is continuously flowing away from the starting field line,
+    (whereas g(α, ζ) has to "decide" at the cut what the next field line is).
```

Comment:
This seems intuitive to me, so I didn't formally prove it. A "proof by example" would be to plot the Fourier Chebyshev series $\theta(\alpha(\vartheta, \zeta), \zeta)$ along a field line. Now since $\theta$ is continuous and $\alpha$ is continuous in $\vartheta, \zeta$, then by varying $\vartheta, \zeta$ along a field line there is guaranteed to be no discontinuities at the cuts (modulo 2pi). Now check if this matches the double Fourier series $\theta(\vartheta, \zeta)$.

Reply:
I tested this example, and it confirmed what I thought, but in hindsight $g = \theta$ is not a good example since it is impossible to approximate $\theta$ with a double fourier series.